client/consensus/manual-seal/src/lib.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A manual sealing engine: the engine listens for rpc calls to seal blocks and create forks
+//! This is suitable for a testing environment.
+
+use consensus_common::{
+	self, BlockImport, Environment, Proposer, BlockCheckParams,
+	ForkChoiceStrategy, BlockImportParams, BlockOrigin,
+	ImportResult, SelectChain,
+	import_queue::{
+		BasicQueue,
+		CacheKeyId,
+		Verifier,
+		BoxBlockImport,
+	},
+};
+use sp_runtime::{
+	traits::{Block as BlockT, Header as HeaderT},
+	generic::BlockId,
+	Justification,
+};
+use sp_blockchain::HeaderBackend;
+use client_api::backend::Backend as ClientBackend;
+use futures::prelude::*;
+use hash_db::Hasher;
+use transaction_pool::{BasicPool, txpool};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+pub mod rpc;
+mod error;
+
+pub use error::Error;
+pub use rpc::{EngineCommand, CreatedBlock};
+
+/// The synchronous block-import worker of the engine.
+pub struct ManualSealBlockImport<I> {
+	inner: I,
+}
+
+impl<I> From<I> for ManualSealBlockImport<I> {
+	fn from(i: I) -> Self {
+		ManualSealBlockImport { inner: i }
+	}
+}
+
+impl<B: BlockT, I: BlockImport<B>> BlockImport<B> for ManualSealBlockImport<I> {
+	type Error = I::Error;
+
+	fn check_block(&mut self, block: BlockCheckParams<B>) -> Result<ImportResult, Self::Error>
+	{
+		self.inner.check_block(block)
+	}
+
+	fn import_block(
+		&mut self,
+		block: BlockImportParams<B>,
+		cache: HashMap<CacheKeyId, Vec<u8>>,
+	) -> Result<ImportResult, Self::Error> {
+		// TODO: strip out post-digest.
```

Comment:
no TODOs

Reply:
actually this was originally your TODO, :sweat_smile:

Reply:
i thought that it was :) but I wouldn't mark a PR as "pleasereview" with an active TODO.

Reply:
Yeah, I understand, will keep that in mind for next time. 

> actually this was originally your TODO

and so what I meant by this is, I have no idea how to resolve this, I would assume to empty the post-digests `Vec` or are there any other additional operations required?

Reply:
If you haven't added a post-digest anywhere during the sealing process then the TODO should just be removed